cli/cmd/main.go

```diff
@@ -883,6 +895,29 @@ func createDataFilter(c *cli.Context) (*datapb.Filter, error) {
 	if len(c.StringSlice(dataFlagMimeTypes)) != 0 {
 		filter.MimeType = c.StringSlice(dataFlagMimeTypes)
 	}
+	if c.StringSlice(dataFlagTags) != nil && c.String(dataFlagTaggedUntagged) != "" {
+		return nil, errors.New("cannot filter based on matching and tagged or untagged")
+	}
+	if c.StringSlice(dataFlagTags) != nil {
+		filter.TagsFilter = &datapb.TagsFilter{
+			Type: datapb.TagsFilterType_TAGS_FILTER_TYPE_MATCH_BY_OR,
+			Tags: c.StringSlice(dataFlagTags),
+		}
+	}
+	if c.String(dataFlagTaggedUntagged) != "" {
```

Comment:
Thinking through how to incorporate tags and made sense to me have separate use cases for 1) filtering based on matching tags and then grabbing either 2) all tagged data or 3) all untagged data.

Very open to alternative ways of going about this